src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java

```diff
@@ -113,6 +114,7 @@ public void prepare(String legalEntityWithUsersResource) {
         } else {
             legalEntityService.setRootAdmin(globalProperties.getString(PROPERTY_ROOT_ENTITLEMENTS_ADMIN));
         }
+        externalStateIds = this.productSummaryConfigurator.ingestArrangementCustomStateAndGetExternalIds();
```

Comment:
is an arrangement state something related to the customer of the bank? i would think the possible arrangement states are independent of the number of customers.

if so you could create the states at once via the `ProductSummaryDataGenerator`, and use the external ids you created/predefined yourself in the existing `ingestArrangements` of `ProductSummaryConfgurator`